src/Mutator/Removal/MethodCallRemoval.php

```diff
@@ -48,6 +48,13 @@ final class MethodCallRemoval implements Mutator
 {
     use GetMutatorName;
 
+    /** @var string[] */
+    private $whitelisted = [
+        'close',
+        'free',
+        'free_result',
```

Comment:
I'm not terribly excited about a special list for instance methods. I wish we could have this either configurable, or completed with the means supposed to exists under #1231

Reply:
If it was configurable you would need this everywhere ? 
How about checking that all of these in the method versions take no argument ? eg `$someClass->free()` would be ignored but `$someClass->free($someVariable)` would be allowed to mutate ?

Reply:
I think doing so against native calls is ok but extending it to arbitrary classes is not: a user will likely be able to test that its `$someClass->free()` is called or not

Reply:
They return void though ? https://www.php.net/manual/en/mysqli-result.free.php or https://www.php.net/manual/en/event.free could check if void return and 0 arguments and called one of these things ?

Reply:
and how do you tell the difference between `$mysqliResult->free()` and `$repository->free()`? We do not have type inference at the moment so we cannot distinguish the two and the later may deserve to be covered by infection

Reply:
I agree there will be methods missed by doing this, I don't have the stats but most methods I have seen called one of these are methods which have no real observable behaviour. 

Could have this as a user config option instead at project level instead of line level ?

Reply:
I don't think a user config option for this list is a bad idea.